include/bitcoin/bitcoin/chain/block.hpp

```diff
@@ -157,19 +158,20 @@ class BC_API block
     bool is_extra_coinbases() const;
     bool is_final(size_t height, uint32_t block_time) const;
     bool is_distinct_transaction_set() const;
-    bool is_valid_coinbase_claim(size_t height) const;
+    bool is_valid_coinbase_claim(size_t height, const settings& settings) const;
```

Comment:
Let's pass the two necessary settings value to `subsidy`, `reward` and `is_valid_coinbase_claim`. This will result in one more parameter for the latter two methods, but I'd like to stick to the approach of not using settings for methods/functions. However we can make an exception for `check`, `accept`, and `connect`.

Reply:
Will do :+1: